src/core/utils.js

```diff
@@ -360,6 +362,27 @@ export function isCompatibleWithUserAgent({
         https://github.com/mozilla/addons-frontend/issues/2074`);
     }
 
+    // Check to see if:
```

Comment:
Why do we need to make a special case for this? We already have code that checks version compatibility. Can't you just re-use that? Here is an idea:

```diff
diff --git a/src/core/utils.js b/src/core/utils.js
index c91213fa..12cf9e29 100644
--- a/src/core/utils.js
+++ b/src/core/utils.js
@@ -310,6 +310,18 @@ export function getCompatibleVersions({ _log = log, addon, clientApp } = {}) {
     if (addon.current_version.compatibility[clientApp]) {
       maxVersion = addon.current_version.compatibility[clientApp].max;
       minVersion = addon.current_version.compatibility[clientApp].min;
+    } else if (
+      addon.type === ADDON_TYPE_EXTENSION &&
+      clientApp === 'firefox' &&
+      (
+        !addon.current_version || !addon.current_version.files ||
+        !addon.current_version.files.length ||
+        addon.current_version.files[0].is_webextension !== true
+      )
+    ) {
+      // This is a legacy extension that is only supported by
+      // Firefox 57+
+      minVersion = '57.0';
     } else if (addon.type === ADDON_TYPE_OPENSEARCH) {
       _log.info(oneLine`addon is type ${ADDON_TYPE_OPENSEARCH}; no
         compatibility info found but this is expected.`, { addon, clientApp });

```

After that, I suspect that you'll get all the comparison code for free. You'll just need to display the error message which might simultaneously fix https://github.com/mozilla/addons-frontend/issues/2087 🍰